### ISSUE TITLE
Sort large data make spyder window stuck

### ISSUE THREAD

**Issue by zysNLP** [spyder-ide/spyder#16133]
## Problem Description

I opened a relatively large data with pandas in spyder, about 100,000 data, and there are more than a dozen columns in it. After I double-clicked this variable, a data window popped up. Some of these columns are strings and some are numeric. When I first opened it, it was more flexible to use the mouse wheel to pull down, and then if I clicked on the column name of a numeric data column, it would sort me from small to large. At this time, I also used the mouse wheel to pull down very flexibly. However, once I click on the column name again to sort the column from largest to smallest, this operation will cause the computer to be very stuck, and it will take a long time to sort the order. At this time, if I use the mouse wheel to scroll, it will become very stuck. . Is this the cause of spyder or pandas?

## Versions

* Spyder version: 5.0.5/5.0.3
* Python version: 3.6.8
* Qt version: 5.9.7
* PyQt version: 5.9.2
* Operating System name/version: win10/ubuntu18.04

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by ccordoba12** [comment]
Hey @zysNLP, thanks for reporting. This is a problem with Spyder and should be related to your amount of available RAM memory.

How much RAM does your computer have?

**Comment by zysNLP** [comment]
@ccordoba12 My computer has 16G RAM with win10 system. Is it too small?

**Comment by ccordoba12** [comment]
Yes, it could be. By scrolling down, you're loading more of your dataframe data into the graphical table that displays it (we load a small fraction when opening it precisely to avoid that).

The thing is we need enough memory to:

1. Create your dataframe in the first place.
2. Open it in Spyder. This is because we need to make a copy of your dataframe to transfer it from the console's kernel (where it lives) to Spyder.
3. Browse it in Spyder. That memory is consumed by the graphical table and it remains small if you don't browse too much of it.

Our plan in the future is to not copy dataframes from the kernel to Spyder by using Pyarrow (which will avoid the waste of memory involved in point 2). But points 1 and 3 are unavoidable.
